onnxruntime/core/mlas/lib/logistic.cpp

```diff
@@ -184,6 +271,10 @@ Return Value:
 #if defined(MLAS_TARGET_AMD64)
     GetMlasPlatform().LogisticKernelRoutine(Input, Output, N);
 #else
-    MlasLogisticKernel(Input, Output, N);
+    #ifdef __ARM_FEATURE_SVE
```

Comment:
Newbie question - So this is set in the arm_sve header if the compiler supports SVE ? Don't we need to check the running processor supports SVE through `HasArmSVE()` before invoking the SVE routine ?

Reply:
Can you please elaborate on the exception you see while running `HasArmSVE()` ? I am not very familiar with the code but it should just check if the processor supports SVE or not and that in itself should not error out (unless there is a bug there that needs to be ironed out).

The issue with the pure compile time check (and not runtime check) is if the MLAS library is built on an SVE supporting platform and run on a platform where it isn't supported. Without a runtime check, wouldn't that cause an issue as well ?

Reply:
`HasArmSVE()` should work on any platform. if it doesn't, we should fix it. but it is just returning a boolean that was set earlier in the CPUIDInfo ctor.

if we compile code conditionally here, we should ensure that it compiles code that can be run. possibly a simpler alternative is to dispatch with the existing function pointer mechanism in MLAS_PLATFORM, where we would only assign the SVE implementation to the function pointer if we know that it can run, instead of directly calling the SVE implementation here.